tox.ini

```diff
@@ -32,27 +36,37 @@ description =
     warnings: treating warnings as errors
     cov: with coverage
     xdist: using parallel processing
+allowlist_externals =
+    git
 pass_env =
     CRDS_*
     HOME
     CI
     TOXENV
     CODECOV_*
+set_env =
+    devdeps: UV_INDEX = https://pypi.anaconda.org/liberfa/simple https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+    devdeps: UV_INDEX_STRATEGY = unsafe-any-match
 extras =
     test
+uv_resolution =
+    oldestdeps: lowest-direct
```

Comment:
This isn't equivalent to `minimum_dependencies`. Testing locally I can reduce the lower numpy to `0.01` and the environment doesn't fail due to `uv` resolving this to a later version of numpy (1.22) and not throwing an error. Would you switch this back to `minimum_dependencies`.

Reply:
No I plan on ending support for minimum_dependencies in the future.

Reply:
I see no mention of any end of life for https://github.com/OpenAstronomy/minimum_dependencies and it's in use by other packages (like [sunpy](https://github.com/sunpy/sunpy/blob/644308816c9ad8443cdf8cc94a9e7fd428286e2e/tox.ini#L71)) that are interested in keeping it alive. Given that it's a better test of our lower pins I say we keep using it.

Reply:
I can do 
```console
uv pip install . --resolution lowest-direct
```
and it attempts to install `numpy==1.22.0`:
```
Resolved 20 packages in 116ms
  × Failed to download and build `numpy==1.22.0`
  ├─▶ Build backend failed to determine requirements with `build_wheel()` (exit
  │   status: 1)

  │   [stderr]
  │   Traceback (most recent call last):
  │     File "<string>", line 8, in <module>
  │       from setuptools.build_meta import __legacy__ as backend
  │     File
  │   "/Users/zburnett/.cache/uv/builds-v0/.tmpy0oCfs/lib/python3.13/site-packages/setuptools/__init__.py",
  │   line 10, in <module>
  │       import distutils.core
  │   ModuleNotFoundError: No module named 'distutils'

  ╰─▶ distutils was removed from the standard library in Python 3.12. Consider
      adding a constraint (like `numpy >1.22.0`) to avoid building a version of
      numpy that depends on distutils.
  help: `numpy` (v1.22.0) was included because `roman-datamodels`
        (v0.22.1.dev4+gab68c35) depends on `numpy`
```

Reply:
Thanks @zacharyburnett is there a way to translate that failure to the CI so that the `oldestdeps` job will fail if our minimum pins are not correct?

With this PR if I set the lower numpy pin to 0.01:
https://github.com/spacetelescope/roman_datamodels/pull/430/commits/22e67d66d0d8bb04d06fc5a6f43cd72998e7ba66
the oldestdeps job installs `1.22`:
https://github.com/spacetelescope/roman_datamodels/actions/runs/12013414624/job/33486895903?pr=430#step:10:46
and passes making that job much less useful.

Reply:
I believe that might be because Python 3.10 only supports `numpy>=1.22.0`, right? So the lowest possible version of Numpy is `1.22.0`, even if you specify `0.01`

Reply:
I don't understand what you mean by

> our minimum pins are not correct

The `numpy` dependency is in the `pyporoject.toml` because `roman_datamodels` directly imports it. The minimum pin `roman_datamodels` lists really only matters for its own uses of `numpy`. If some upstream dependency raises its lower pin that does not really matter to `roman_datamodels` as any install should be the transitive closure the dependencies, which is what `uv` is doing, only it targets the lowest version rather than maximum version of each dependency in this case.

Note, I did not write `minimum_dependencies` so that it would exactly "check minimum" pins in the way you are implying, namely that they are exactly correct. I wrote it so that one could check a minimum dependency version install passed the tests. This was so that we could identify via test failures the minimum versions of dependencies within the package being tested.

What you are referencing seems to indicate that you want to test that the minimum dependency listed within a given package is `>=` the transitive closure of the minimum pins in all the other dependencies. I don't think this is particularly useful considering any install should be done via a valid transitive closure rather than some exact pinning scheme. All the exact pinning scheme does is create work for your package due to pinning changes upstream. The closure should take care of finding the minimum valid version, the package's pin itself should only serve to indicate the minimum version its own internal code enforces.

Note `minimum_dependencies` only relies on the exact pinning scheme because I did want to implement and support the algorithms necessary to determine the transitive closure of the minimum versions for all the dependencies in general.